pensieve/pensieve.py

```diff
@@ -146,10 +146,22 @@ def paragraphs(self):
         if not self._paragraphs:
             print('Generating paragraphs for doc '+str(self.id))
             self._paragraphs = []
-            for i, p in enumerate(self.text.split('\n')):
-                if len(p.split(' ')) < 30:
+            list = self.text.split('\n')
+            i = 0
+            while i < len(list):
+                if len(list[i].strip()) == 0:
+                    i += 1
                     continue
-                self._paragraphs.append(Paragraph(p, i, self))
+                if len(list[i].split(' ')) >= 25:
+                    self._paragraphs.append(Paragraph(list[i], i, self))
+                else:
+                    chunk = ""
+                    while (i < len(list)) and (len(list[i].split(' ')) < 25 or list[i][0] == "'" or list[i][0] == '"'):
```

Comment:
So my logic for using a while loop was that I _wanted_ to check (and modify) where I was in the loop. For example, if I took several lines of dialogue and combined it into one Paragraph, I didn't want the next Paragraph to start off with the second line of dialogue, I wanted it to start with the text after the dialogue.